curator/actions.py

```diff
@@ -1303,6 +1303,36 @@ def _post_run_quick_check(self, index_name):
                 'not found.'.format(index_name)
             )
 
+        # Verify that the reindex task finished without errors
+        response = self.client.tasks.get(task_id)
+        failures = response['response']['failures']
+
+        if len(failures) > 0:
+            self.loggit.error(
+                'The reindex task completed with {0} errors. '
+                'Check task: {1} for more information.'.format(len(failures), task_id)
+            )
+
+            raise exceptions.FailedExecution(
+                'Reindex completed with {0} errors. '
+                'Check task: {1} for more information.'.format(len(failures), task_id)
+            )
+
+        # Verify that the number of docs in the source and destination index match
+        total_docs = response['task']['status']['total']
+        created_docs = response['task']['status']['created']
+
+        if not created_docs == total_docs:
```

Comment:
So, devil's advocate here.  What happens if someone does a reindex from query instead of a full index? Will these still be equal? What if someone reindexes from multiple indices to 1 index (e.g. 1 week's worth of daily indices into a single  weekly index)? Do these numbers match up? If the answer is not "yes," then I can't use the code—at least not as-is.

The draft code works just fine in the event of a 1:1 reindex. I could merge it if the test is only made made when a `verify_doc_count` flag is present (or something like it).  It would, of course, have to be vetted by conditional tests.  It wouldn't be valid, for example, if Curator was able to detect a many-to-one reindex (code which would have to be added).

Reply:
If you reindex going from a mapping file without nested documents to a mapping file with nested documents, the document count will be different.  I assume the doc count verification should be optional...

Reply:
Thanks for providing a concrete example, @rahst12